apps/cli/src/commands/build.ts

```diff
@@ -0,0 +1,296 @@
+        if (imageInfo["Architecture"] !== "riscv64") {
+            throw new Error(
+                `Invalid image Architecture: ${imageInfo["Architecture"]}. Expected riscv64`
+            );
+        }
+
+        const labels = imageInfo["Config"]["Labels"] || {};
+        const info: ImageInfo = {
+            cmd: imageInfo["Config"]["Cmd"] ?? [],
+            dataSize: labels[CARTESI_LABEL_DATA_SIZE] ?? "10Mb",
+            entrypoint: imageInfo["Config"]["Entrypoint"] ?? [],
+            env: imageInfo["Config"]["Env"] || [],
+            ramSize: labels[CARTESI_LABEL_RAM_SIZE] ?? DEFAULT_RAM_SIZE,
+            sdkVersion:
+                labels[CARTESI_LABEL_SDK_VERSION] ?? DEFAULT_SDK_VERSION,
+            workdir: imageInfo["Config"]["WorkingDir"],
+        };
+
+        if (!info.entrypoint && !info.cmd) {
+            throw new Error("Undefined image ENTRYPOINT or CMD");
+        }
+
+        // warn for using default values
+        info.sdkVersion ||
+            this.warn(
+                `Undefined ${CARTESI_LABEL_SDK_VERSION} label, defaulting to ${DEFAULT_SDK_VERSION}`
+            );
+
+        info.ramSize ||
+            this.warn(
+                `Undefined ${CARTESI_LABEL_RAM_SIZE} label, defaulting to ${DEFAULT_RAM_SIZE}`
+            );
+
+        // validate data size value
+        if (bytes(info.dataSize) === null) {
+            throw new Error(
+                `Invalid ${CARTESI_LABEL_DATA_SIZE} value: ${info.dataSize}`
+            );
+        }
+
+        // XXX: validate other values
+
+        return info;
+    }
+
+    private async exportImageTar(
+        image: string,
+        tarPath: string
+    ): Promise<void> {
+        // create container
+        const { stdout: cid } = await execa("docker", [
+            "create",
+            "--platform",
+            "linux/riscv64",
+            image,
+        ]);
+
+        // export container rootfs to tar
+        await execa("docker", ["export", "-o", tarPath, cid]);
+
+        // remove container
+        await execa("docker", ["rm", cid]);
+    }
+
+    private async createExt2(
+        toolchainImage: string,
+        info: ImageInfo,
+        tarPath: string
+    ): Promise<string> {
+        // extract base name of tar file
+        const tarName = path.basename(tarPath, path.extname(tarPath));
+
+        const containerDir = "/mnt";
+        const bind = `${path.resolve(path.dirname(tarPath))}:${containerDir}`;
+        const tar = path.join(containerDir, tarName + ".tar");
+        const ext2 = path.join(containerDir, tarName + ".ext2");
+
+        // calculate extra size
+        const blockSize = 4096;
+        const extraBytes = bytes.parse(info.dataSize);
+        const extraBlocks = Math.ceil(extraBytes / blockSize);
+        const extraSize = `+${extraBlocks}`;
+
+        await execa(
+            "docker",
+            [
+                "run",
+                "--rm",
+                "-v",
```

Comment:
Let's favor the long version for clarity.

```suggestion
                "--volume",
```